### PR TITLE
MWPW-172799: Graybox Support for CaaS

### PR DESCRIPTION
**Pull request by shkhan91**
# MWPW-172799: Graybox Support for CaaS

## Overview

This PR introduces **Graybox Experience ID support for CaaS**. The changes ensure that when a page is served from a graybox domain, the experience ID is correctly detected and included in the CaaS card metadata. This enables more accurate experience mapping for graybox environments, which will be used for executive reviews.

## Key Features & Fixes

- **Graybox Experience ID Extraction:**  
  - Added a utility (`getGrayboxExperienceId`) that extracts the experience ID from both standard graybox domains (e.g., `https://myexp.us-graybox.adobe.com/`) and stage graybox domains (e.g., `https://stage--us-graybox-adobecom.aem.page/myexp/`).
  - The logic is robust and handles malformed or non-graybox URLs gracefully.

- **CaaS Metadata Enhancement:**  
  - The CaaS card metadata now includes a `gbExperienceID` property when a graybox experience is detected.
  - This property is automatically set based on the current page URL or a provided URL.

- **Bulk Publisher Support:**  
  - The bulk publishing tool now passes the page URL to the CaaS property builder, ensuring graybox experience IDs are included during bulk operations.

- **Comprehensive Testing:**  
  - Added and updated unit tests to cover all graybox detection scenarios, including edge cases and fallbacks.
  - Tests verify that the `gbExperienceID` is set correctly for all supported URL formats and is omitted for non-graybox domains.

## Resolves

[MWPW-172799](https://jira.corp.adobe.com/browse/MWPW-172799)

## Test URLs

- **Before:**  
  https://main--milo--adobecom.aem.page/?martech=off
- **After:**  
  https://MWPW-172799-stage--milo--adobecom.aem.page/?martech=off

### CODE REVIEW

File: libs/blocks/caas/utils.js [hunk 1]
```diff
@@ -647,6 +647,48 @@ export const stageMapToCaasTransforms = (config) => {
...
+ * @param {string} [pathname] - Optional pathname, defaults to window.location.pathname
+ * @returns {string|null} The experience ID or null if not found
+ */
+export const getGrayboxExperienceId = (
+  hostname = window.location.hostname,
+  pathname = window.location.pathname,
+) => {
+  // Only allow trusted Adobe graybox domains
+  const isAdobeGraybox = (
+    hostname.endsWith('.adobe.com')
```
Comment (chrischrischris) [inline libs/blocks/caas/utils.js:665]: Is this check needed with the regex check below?

File: libs/blocks/caas/utils.js [hunk 2]
```diff
@@ -669,6 +711,10 @@ export const getConfig = async (originalState, strs = {}) => {
 
   const caasRequestHeaders = addFloodgateHeader(state);
 
+  // Get graybox experience ID if on graybox domain
```
Comment (chrischrischris) [inline libs/blocks/caas/utils.js:714]: nit: comment not needed

**Review by gmirijan — APPROVED** [review]

**Review by sheridansunier — APPROVED** [review]
